src/main/java/org/swisspush/redisques/action/AbstractQueueAction.java

```diff
@@ -60,9 +61,8 @@ protected Handler<Throwable> replyErrorMessageHandler(Message<JsonObject> event)
         };
     }
 
-    protected void handleFail(Message<JsonObject> event, String message, Throwable throwable) {
-        log.warn(message, exceptionFactory.newException(throwable));
-        event.fail(0, throwable.getMessage());
+    protected void handleFail(Message<JsonObject> event, String message, Throwable cause) {
+        event.reply(exceptionFactory.newReplyException(RECIPIENT_FAILURE, 0, null, cause));
```

Comment:
You could maybe hide parameters like `RECIPIENT_FAILURE` and `0` in a dedicated method in the `exceptionFactory`.

Something like
```java
event.reply(exceptionFactory.newRecipientFailureReplyException(null, cause));
```

Reply:
I like this idea :) I'll have a look.